### ISSUE TITLE
Getting Response as "{{0xc4201263f0} <nil> <nil> <nil> <nil> <nil> <nil> map[]}"

### ISSUE THREAD

**Issue by Ranjith9** [Azure/azure-sdk-for-go#1726]
Hi,

I am creating Resources from Compute and Network. The response I am getting is like 

" root@machine:~# go run  create.go
![image](https://user-images.githubusercontent.com/24772433/39417811-62f5cb2c-4c74-11e8-87d7-286ff1e43139.png) "

Here is my code for creating Microsoft.Network

func main() {
        vnetClient := network.NewVirtualNetworksClient(subscriptionID)
        vnetClient.Authorizer = autorest.NewBearerAuthorizer(token)

        response, err := vnetClient.CreateOrUpdate(ctx, "test-rg", "test",
                network.VirtualNetwork{
                        Location: to.StringPtr("CentralIndia"),
                        VirtualNetworkPropertiesFormat: &network.VirtualNetworkPropertiesFormat{
                                AddressSpace: &network.AddressSpace{
                                        AddressPrefixes: &[]string{"192.168.0.0/16"},
                                },
                        },
                })

        err = response.WaitForCompletion(ctx, vnetClient.Client)
        if err != nil {
                fmt.Errorf("cannot get the vnet create or update future response: %v", err)
        }
        fmt.Println(response)
}

Previously I was getting the response of VirtualNetwork struct with the same code.

**Comment by jhendrixMSFT** [comment]
What was the previous version of  the SDK you were using?  We've made some changes to how long-running async operations are handled, specifically they now return a future and you need to ask it for the result.  From your code sample I think you need to do the following.
```golang
err = response.WaitForCompletion(ctx, vnetClient.Client)
// error handling...
fmt.Println(response.Result(vnetClient))
```
Let me know how this works for you.

**Comment by Ranjith9** [comment]
Thanks for the reply.

I have also tried the above code you mentioned. In fact I have tried the examples from the https://github.com/Azure-Samples/azure-sdk-for-go-samples, specifically "CreateVirtualNetwork" function in network package.

func CreateVirtualNetwork(ctx context.Context, vnetName string) (vnet network.VirtualNetwork, err error) {
	vnetClient := getVnetClient()
	future, err := vnetClient.CreateOrUpdate(
		ctx,
		helpers.ResourceGroupName(),
		vnetName,
		network.VirtualNetwork{
			Location: to.StringPtr(helpers.Location()),
			VirtualNetworkPropertiesFormat: &network.VirtualNetworkPropertiesFormat{
				AddressSpace: &network.AddressSpace{
					AddressPrefixes: &[]string{"10.0.0.0/8"},
				},
			},
		})

	if err != nil {
		return vnet, fmt.Errorf("cannot create virtual network: %v", err)
	}

	err = future.WaitForCompletion(ctx, vnetClient.Client)
	if err != nil {
		return vnet, fmt.Errorf("cannot get the vnet create or update future response: %v", err)
	}

	return future.Result(vnetClient)
}

Previously I was using "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2017-09-01/network". Even this version is giving the same output now.

The Operation is success while creating resources (Network and Compute) but the response I'm getting is nil.

**Comment by jhendrixMSFT** [comment]
I suspect this is the same issue as https://github.com/Azure/azure-sdk-for-go/issues/1462.  This should be fixed in the v17 release which should be released this week.

**Comment by jhendrixMSFT** [comment]
This should now be fixed in v17.0.0 of the SDK.  Please re-open if you're still seeing this problem with this version.
https://github.com/Azure/azure-sdk-for-go/releases/tag/v17.0.0
